### PR TITLE
Dump all Kubernetes resources in dedicated file

### PR DESCRIPTION
**Pull request by JRBANCEL**
**What this PR does, why we need it**:
Follow up on #1352:
* `kubectl get all` doesn't dump the CRDs and doesn't even dump all Kubernetes resources, so it is replaced by a manual loop dumping everything in a clean fashion (high level listing first and then full dump)
* Removed `kubectl get ...` calls that are now redundant
* Dump all of this in a dedicated file to keep `build.log` clean instead of immensely bloated as it is today

### CODE REVIEW

File: scripts/e2e-tests.sh [hunk 1]
```diff
@@ -93,15 +93,30 @@ function dump_cluster_state() {
...
-  echo ">>> Events:"
-  kubectl get events --all-namespaces --show-labels -o wide
-  function_exists dump_extra_cluster_state && dump_extra_cluster_state
-  echo ">>> Full dump: ${ARTIFACTS}/k8s.dump.txt"
-  kubectl get all --all-namespaces -o yaml > ${ARTIFACTS}/k8s.dump.txt
+
+  local output="${ARTIFACTS}/tmp/k8s.dump.txt"
+  echo ">>> The dump is located at ${output}"
+
+  for crd in `kubectl api-resources --verbs=list -o name | sort`; do
```
Comment (adrcunha) [inline scripts/e2e-tests.sh]: $() is preferred over `` for readability sake.

```suggestion
  for crd in $(kubectl api-resources --verbs=list -o name | sort); do
```

File: scripts/e2e-tests.sh [hunk 2]
```diff
@@ -93,15 +93,30 @@ function dump_cluster_state() {
...
+    count=$(kubectl get $crd --all-namespaces --no-headers 2>/dev/null | wc -l)
```
Comment (adrcunha) [inline scripts/e2e-tests.sh]:
```suggestion
    local count="$(kubectl get $crd --all-namespaces --no-headers 2>/dev/null | wc -l)"
```

File: scripts/e2e-tests.sh [hunk 3]
```diff
@@ -93,15 +93,30 @@ function dump_cluster_state() {
...
-  kubectl get all --all-namespaces --show-labels -o wide
-  echo ">>> Services:"
-  kubectl get services --all-namespaces --show-labels -o wide
-  echo ">>> Events:"
-  kubectl get events --all-namespaces --show-labels -o wide
-  function_exists dump_extra_cluster_state && dump_extra_cluster_state
-  echo ">>> Full dump: ${ARTIFACTS}/k8s.dump.txt"
-  kubectl get all --all-namespaces -o yaml > ${ARTIFACTS}/k8s.dump.txt
+
+  local output="${ARTIFACTS}/tmp/k8s.dump.txt"
```
Comment (chaodaiG) [inline scripts/e2e-tests.sh]: I love this idea of not bloating e2e logs!
Don't you need to create the `${ARTIFACTS}/tmp` dir before dumping logs?

Comment (JRBANCEL) [reply scripts/e2e-tests.sh]: /tmp is a bug, is is an artifact from my local debugging.
Good catch, will fix.

Comment (adrcunha) [reply scripts/e2e-tests.sh]: Maybe applying this suggestion is easier? :)

```suggestion
  local output="${ARTIFACTS}/k8s.dump.txt"
```

File: scripts/e2e-tests.sh [hunk 4]
```diff
@@ -93,15 +93,30 @@ function dump_cluster_state() {
...
+      if [[ "${crd}" == "secrets" ]]; then
+        echo "Secrets are ignored for security reasons" >> ${output}
+      else
+        kubectl get ${crd} --all-namespaces -o yaml >> ${output}
+      fi
+    fi
+  done
+
+  echo ">>> Extra dump" >> ${output}
+  function_exists dump_extra_cluster_state && dump_extra_cluster_state >> ${output}
```
[the remark this answers is not in the document]
Comment (JRBANCEL) [reply scripts/e2e-tests.sh]: Makes sense. Done.

File: scripts/e2e-tests.sh [hunk 5]
```diff
@@ -93,15 +93,29 @@ function dump_cluster_state() {
...
+      echo ">>> Details" >> ${output}
+      if [[ "${crd}" == "secrets" ]]; then
+        echo "Secrets are ignored for security reasons" >> ${output}
+      else
+        kubectl get ${crd} --all-namespaces -o yaml >> ${output}
+      fi
+    fi
+  done
+
+  function_exists dump_extra_cluster_state && echo ">>> Extra dump" >> ${output} && dump_extra_cluster_state >> ${output}
```
Comment (adrcunha) [inline scripts/e2e-tests.sh]: Please use `if` here for readability.

**Comment by adrcunha** [comment]
Thanks!

/approve
/lgtm

**Pull request merged**